fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -566,10 +566,12 @@ public OptExpression rewrite() {
         final ReplaceColumnRefRewriter mvColumnRefRewriter =
                 MvUtils.getReplaceColumnRefWriter(mvExpression, mvColumnRefFactory);
 
-        final List<ScalarOperator> mvConjuncts = MvUtils.getAllValidPredicates(mvExpression);
+        final List<ScalarOperator> mvConjuncts = MvUtils.getAllValidPredicates(mvExpression, optimizerContext);
         ScalarOperator mvPartitionCompensate = compensateMVPartitionPredicate(mvConjuncts, mvColumnRefRewriter);
         if (mvPartitionCompensate != ConstantOperator.TRUE) {
-            mvConjuncts.addAll(MvUtils.getAllValidPredicates(mvPartitionCompensate));
+            Utils.extractConjuncts(mvPartitionCompensate).stream()
+                    .filter(p -> !optimizerContext.getPushdownDerivedPredicates().contains(p))
```

Comment:
Why not push this into `getAllValidPredicates ` method?

Reply:
fixed.